sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment.py

```diff
@@ -248,6 +261,21 @@ def _from_rest_object(cls, deployment: BatchDeploymentData):  # pylint: disable=
             properties=deployment.properties.properties,
         )
 
+        # Job definition is in private preview. If private preview environment is
+        # not enable we need to remove job definition from properties.
+        if not is_private_preview_enabled():
+            del_key = []
+            for k in deployment.properties:
+                if k.startswith("component_deployment"):
```

Comment:
Regardless of the key name (one of my other comments) please make this a constant